beetsplug/bandcamp/__init__.py

```diff
@@ -218,9 +218,11 @@ def candidates(
 
         if album == "":
             search = {
-                "query": artist + " - " + item.title,
-                "artist": artist,
-                "label": label,
+                "query": item.artist + " - " + item.title
+                if item.artist != ""
+                else item.title,
+                "artist": item.artist,
+                "label": item.label,
```

Comment:
```suggestion
                "label": label,
```

Not sure whether this change was intentional - see the logic above where you will find how the label is set. We do not want to adjust it here.

Reply:
thanks overlooked that one!